### PR TITLE
Split the CI pipelines

### PR DESCRIPTION
**Pull request by raymondxyang**
Split the CI pipelines.

Steps:
1. Merge the new configuration into master (Not enforced on CI)
2. Enable contiguous build on master branch, update the build badge in README
3. After a while (all PR synced with master and has the new CI configuration (targeting end of the week)) switch to enforce the new separated pipelines and remove the existing pipeline configuration.

### CODE REVIEW

**Comment by pranavsharma** [comment]
LGTM. Will this require 5 badges on the README?

**Comment by pranavsharma** [comment]
@jywu-msft fyi

**Comment by raymondxyang** [comment]
yes. I already wrote a CI status table in readme. Will send a PR after this one is in

File: tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+jobs:
+- job: Linux_CI_Dev
+  pool: Linux-CPU
+  steps:
+    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o ubuntu16.04 -d cpu -r $(Build.BinariesDirectory) -x "--use_mklml"'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml]: not specific to this PR, but remind me what was the reasoning to have some options in the yml and some options in the build script? (e.g. --enable_mkldnn is in run_build.sh but --use_mklml is here)

Comment (raymondxyang) [reply tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml]: run_build.sh has the minimal set of the parameters.. we want to test builds with mklml, but dont want it when we packaging the releases (in order to reuse the script)

**Pull request merged**